operator/api/redpanda/v1alpha2/redpanda_types.go

```diff
@@ -292,7 +292,20 @@ func (in *Redpanda) GetDot(restConfig *rest.Config) (*helmette.Dot, error) {
 	var values []byte
 	var partial redpandachart.PartialValues
 
-	values, err := json.Marshal(in.Spec.ClusterSpec)
+	spec := in.Spec.ClusterSpec.DeepCopy()
```

Comment:
This would fit better into a helper method. It's strange to have to call `GetDot` only to discard 50% of what it's doing.

You could move this into an `AsValues` method or try to be fancy and put it in to `MarshalJSON`